### PR TITLE
[iOS] Add missing call to ReloadData on NotifyCollectionChangedAction.Reset

### LINKED ISSUE #5793: [CollectionView/ListView] Not listening for Reset command
### Description
See issue: https://github.com/jamesmontemagno/mvvm-helpers/issues/39

### Basic Information
Xamarin.Forms 3.6.0.264807
Visual Studio 2017 Version 15.9.10
Platforms:
iOS Simulator with iOS 12.1 (iPhone SE)

### Reproduction Link
https://github.com/xamadev/ObservableRangeCollectionError

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

The iOS adapter for INotifyCollectionChanged is missing a call to `ReloadData` when handling `NotifyCollectionChangedAction.Reset`. This adds the missing call.

### Issues Resolved ### 

- fixes #5793 

### API Changes ###

None

### Platforms Affected ### 

- iOS

### CODE REVIEW

**Comment by hartez** [comment]
Adding and DNM for the moment - I want to add a UI test so we can make sure this doesn't pop up again.

File: Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs [hunk 1]
```diff
@@ -59,6 +59,7 @@ void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 					Move(args);
 					break;
 				case NotifyCollectionChangedAction.Reset:
+					_collectionView.ReloadData();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs]: failing on IOS10 :(

Comment (abdul-wasey) [reply Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs:62]: Still not working.. XF Version = 4.8.0.1687

**Comment by abdul-wasey** [comment]
Still not working..
XF Version = 4.8.0.1687

**Pull request merged**